### PR TITLE
Instance lifecycle on starting

### PR DESCRIPTION
**Pull request by MiguelMoll**
Taking over PR #1389

### CODE REVIEW

**Comment by MiguelMoll** [comment]
[Lifecycle on instance creation](https://trello.com/c/PrRvjzTt/1151-lifecycle-on-instance-creation)

File: provider/aws/dist/rack.json [hunk 1]
```diff
@@ -989,7 +991,7 @@
             "bootcmd:\n",
             "  - mkswap /dev/xvdb\n",
             "  - swapon /dev/xvdb\n",
-            "  - yum install -y nfs-utils\n",
+            "  - yum install -y nfs-utils aws-cli\n",
```
Comment (mwarkentin) [inline provider/aws/dist/rack.json]: Organize these alphabetically, or is that too OCD? :)

File: provider/aws/dist/rack.json [hunk 2]
```diff
@@ -1094,7 +1100,18 @@
         }
       }
     },
-    "InstancesLifecycle": {
+    "InstancesLifecycleStarting": {
```
Comment (mwarkentin) [inline provider/aws/dist/rack.json]: Maybe `InstancesLifecycleLaunching` instead to match the official AWS naming convention?

**Comment by mwarkentin** [comment]
Code LGTM. Will follow up with testing results on our staging rack tomorrow.

**Comment by mwarkentin** [comment]
Confirmed that this is functioning as expected on our staging rack:

**Comment by mwarkentin** [comment]
During the update, no instances were terminated by the monitor:

```
~wave/stage~ $ convox rack logs --since 35m --filter why
2016-11-15T16:44:51Z web:20161114200708-instance-lifecycle-starting/6a3fb463bdab who="EC2/ASG" what="Launching a new EC2 instance: i-0400d0f05e5bb887c" why="At 2016-11-15T16:41:14Z a user request update of AutoScalingGroup constraints to min: 5, max: 1000, desired: 5 changing the desired capacity from 4 to 5.  At 2016-11-15T16:41:42Z an instance was started in response to a difference between desired and actual capacity, increasing the capacity from 4 to 5."
2016-11-15T16:54:51Z web:20161114200708-instance-lifecycle-starting/6a3fb463bdab who="EC2/ASG" what="Launching a new EC2 instance: i-0c01f91b557c523ad" why="At 2016-11-15T16:50:50Z an instance was started in response to a difference between desired and actual capacity, increasing the capacity from 4 to 5."
2016-11-15T16:59:53Z web:20161114200708-instance-lifecycle-starting/14dfd110963a who="EC2/ASG" what="Launching a new EC2 instance: i-0b0227d0d4be03c5b" why="At 2016-11-15T16:57:57Z an instance was started in response to a difference between desired and actual capacity, increasing the capacity from 4 to 5."
2016-11-15T17:04:53Z web:20161114200708-instance-lifecycle-starting/14dfd110963a who="EC2/ASG" what="Launching a new EC2 instance: i-032e2cd3686f3c2ef" why="At 2016-11-15T17:04:02Z an instance was started in response to a difference between desired and actual capacity, increasing the capacity from 4 to 5."
2016-11-15T17:09:53Z web:20161114200708-instance-lifecycle-starting/14dfd110963a who="EC2/ASG" what="Terminating EC2 instance: i-05a877f7c010752fd" why="At 2016-11-15T17:09:39Z instance i-05a877f7c010752fd was taken out of service in response to a user request, shrinking the capacity from 5 to 4."
2016-11-15T17:17:06Z web:20161114200708-instance-lifecycle-starting/fb1daaf2e9b1 who="EC2/ASG" what="Terminating EC2 instance: i-0400d0f05e5bb887c" why="At 2016-11-15T17:10:07Z instances were launched to balance instances in zones  us-east-1e with other zones resulting in more than desired number of instances in the group.  At 2016-11-15T17:15:42Z an instance was taken out of service in response to a difference between desired and actual capacity, shrinking the capacity from 5 to 4.  At 2016-11-15T17:15:43Z instance i-0400d0f05e5bb887c was selected for termination."
```

**Pull request closed without merging**